optimum/habana/transformers/training_args.py

```diff
@@ -607,10 +625,11 @@ def _setup_devices(self) -> "torch.device":
             from .modeling_utils import adapt_transformers_to_gaudi
 
             adapt_transformers_to_gaudi()
-
             if self.use_lazy_mode:
+                assert (os.getenv('PT_HPU_LAZY_MODE') == '1' or os.getenv('PT_HPU_LAZY_MODE') == None), \
+                    "Argument --use_lazy_mode used, but PT_HPU_LAZY_MODE={os.getenv('PT_HPU_LAZY_MODE')}. For lazy mode, set PT_HPU_LAZY_MODE to 1"
                 logger.info("Enabled lazy mode.")
-            else:
+            elif not self.torch_compile:
                 os.environ["PT_HPU_LAZY_MODE"] = "2"
```

Comment:
Now that there is assert in use_lazy_mode, we no longer can work with new eager mode. Please change this code to do the following:
```
if os.getenv("PT_HPU_LAZY_MODE", "1") != "0":
    os.environ["PT_HPU_LAZY_MODE"] = "2"
```